Assets/HoloToolkit-Examples/GazeRuler/Scripts/ModeTip.cs

```diff
@@ -1,67 +1,73 @@
+
+using UnityEngine;
 using HoloToolkit.Unity;
 
-/// <summary>
-/// provide a tip text of current measure mode
-/// </summary>
-public class ModeTip : Singleton<ModeTip>
+namespace HoloToolkit.Examples.GazeRuler
 {
-    private const string LineMode = "Line Mode";
-    private const string PloygonMode = "Geometry Mode";
-    private TextMesh text;
-    private int fadeTime = 100;
-
-    void Start()
+    /// <summary>
+    /// provide a tip text of current measure mode
+    /// </summary>
+    public class ModeTip : Singleton<ModeTip>
     {
-        text = GetComponent<TextMesh>();
-        switch (MeasureManager.Instance.Mode)
-        {
-            case GeometryMode.Line:
-                text.text = LineMode;
-                break;
-            default:
-                text.text = PloygonMode;
-                break;
-        }
-    }
+        private const string LineMode = "Line Mode";
+        private const string PloygonMode = "Geometry Mode";
+        private TextMesh text;
+        private int fadeTime = 100;
 
-    // Update is called once per frame
-    void Update()
-    {
-        if (gameObject.activeInHierarchy)
+        void Start()
         {
-            // if you want log the position of mode tip text, just uncomment it.
-            // Debug.Log("pos: " + gameObject.transform.position);
+            text = GetComponent<TextMesh>();
             switch (MeasureManager.Instance.Mode)
             {
                 case GeometryMode.Line:
-                    if (!text.text.Contains(LineMode))
-                        text.text = LineMode;
+                    text.text = LineMode;
                     break;
                 default:
-                    if (!text.text.Contains(PloygonMode))
-                        text.text = PloygonMode;
+                    text.text = PloygonMode;
                     break;
             }
+        }
 
-            var render = GetComponent<MeshRenderer>().material;
-            fadeTime = 100;
-            // fade tip text
-            if (fadeTime == 0)
+        // Update is called once per frame
+        void Update()
+        {
+            if (gameObject.activeInHierarchy)
             {
-                var color = render.color;
+                // if you want log the position of mode tip text, just uncomment it.
+                // Debug.Log("pos: " + gameObject.transform.position);
+                switch (MeasureManager.Instance.Mode)
+                {
+                    case GeometryMode.Line:
+                        if (!text.text.Contains(LineMode))
+                            text.text = LineMode;
+                        break;
+                    default:
+                        if (!text.text.Contains(PloygonMode))
+                            text.text = PloygonMode;
+                        break;
+                }
+
+                var render = GetComponent<MeshRenderer>().material;
```

Comment:
This causes a memory leak, unless cleaned up properly.
Please cache the material reference and delete in OnDestroy.
See https://github.com/Microsoft/HoloToolkit-Unity/pull/424 for examples.
Also: nit: naming is incorrect.